### DISCUSSION TITLE
dryRun validation is not checking for `null`. Even though the docs say that `null` is the default.

### DISCUSSION THREAD

**Discussion by frankbo** [renovatebot/renovate#27614] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Base on the Docs it is fine to have `null` as a config value for dryRun
https://docs.renovatebot.com/self-hosted-configuration/#dryrun

When I set it to `null` in our config we get an error message, because the dryRun validation does not permit a `null` value in the config.

https://github.com/renovatebot/renovate/blob/main/lib/config/validation.ts#L850

It might be good to either allow `null` in the config or adjust the docs, that the default has no `dryrun` key at all.

In version 37.214.3 it was still working to have `dryRun: null`. Afterwards the validation was added.

@RahulGautamSingh . Just tagged you here, since you added the validation.

### Relevant debug logs

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by rarkins** [comment]
Converted to high priority bug issue: https://github.com/renovatebot/renovate/issues/27615
